src/components/cc-addon-linked-apps/cc-addon-linked-apps.js

```diff
@@ -8,51 +8,48 @@ import { i18n } from '../../lib/i18n.js';
 import { skeletonStyles } from '../../styles/skeleton.js';
 import { ccLink, linkStyles } from '../../templates/cc-link/cc-link.js';
 
-/** @type {Application[]} */
+/** @type {LinkedApplication[]} */
 const SKELETON_APPLICATIONS = [
-  { name: '??????????????????', link: '', instance: { variant: {} } },
-  { name: '???????????????????????', link: '', instance: { variant: {} } },
-  { name: '????????????????????', link: '', instance: { variant: {} } },
+  { name: '??????????????????', link: '' },
+  { name: '???????????????????????', link: '' },
+  { name: '????????????????????', link: '' },
 ];
 
 /**
- * @typedef {import('../common.types.js').Application} Application
+ * @typedef {import('./cc-addon-linked-apps.types.js').AddonLinkedAppsState} AddonLinkedAppsState
  */
 
 /**
  * A component to display applications linked to an add-on.
  *
- * ## Details
- *
- * * When applications is nullish, a skeleton screen UI pattern is displayed (loading hint).
- *
  * @cssdisplay block
  */
 export class CcAddonLinkedApps extends LitElement {
 
   static get properties () {
     return {
-      applications: { type: Array },
-      error: { type: Boolean },
+      applications: { type: Object },
     };
   }
 
   constructor () {
     super();
 
-    /** @type {Application[]|null} Sets the linked applications. */
-    this.applications = null;
-
-    /** @type {boolean} Displays an error message. */
-    this.error = false;
+    /** @type {AddonLinkedAppsState|null} Sets the linked applications state. */
+    this.applications = { state: 'loading' };
   }
 
   render () {
 
-    const skeleton = (this.applications == null);
-    const applications = skeleton ? SKELETON_APPLICATIONS : this.applications;
-    const hasData = (!this.error && (applications.length > 0));
-    const emptyData = (!this.error && (applications.length === 0));
+    if (this.applications.state === 'error') {
+      return html`
+        <cc-notice intent="warning" message="${i18n('cc-addon-linked-apps.loading-error')}"></cc-notice>
+      `;
+    }
+
+    const skeleton = (this.applications.state === 'loading');
+    const linkedApps = skeleton ? SKELETON_APPLICATIONS : this.applications.linkedApplications;
+    const hasData = (linkedApps.length > 0);
```

Comment:
question: I wonder if going for something like this would help the typechecker:

```
const linkedApps = (this.applications.state === 'loaded') : this.applications.linkedApplications : SKELETON_APPLICATIONS;
const hasData = (linkedApps.length > 0);
```

Reply:
I don't understand:
* I need the `skeleton` const somewhere else.
* I can't see where this will help the typechecker.